ko/news/_posts/2021-12-25-ruby-3-1-0-released.md

```diff
@@ -0,0 +1,257 @@
+
+YJIT는 아직 실험적인 기능이기 때문에 기본으로는 비활성화되어있습니다. 만약 사용하고 싶다면 `--yjit` 명령어 옵션을 통해 활성화할 수 있습니다. 그리고 아직은 Unix-like한 x86-64 플랫폼으로만 한정되어있습니다.
+
+* <https://bugs.ruby-lang.org/issues/18229>
+* <https://shopify.engineering/yjit-just-in-time-compiler-cruby>
+* <https://www.youtube.com/watch?v=PBVLf3yfMs8>
+
+## debug gem: 새로운 디버거
+
+완전히 새로 쓰인 디버거인 [debug.gem](https://github.com/ruby/debug)이 포함됩니다. debug.gem은 다음 기능을 갖고 있습니다:
+
+* 디버깅 성능 향상 (디버거가 있어도 애플리케이션을 느리게 만들지 않음).
+* 원격 디버깅 지원.
+* 풍부한 프론트엔드 디버깅 지원 (이제 VSCode와 크롬 브라우저를 지원).
+* 멀티 프로세스/멀티 스레드 디버깅 지원.
+* REPL 색상 지원.
+* 그리고 녹화 & 재생 기능, 추적 기능 등 다양하고 유용한 기능들.
+
+<video autoplay="autoplay" controls="controls" muted="muted" width="764" height="510" poster="https://cache.ruby-lang.org/pub/media/ruby310_debug_demo.jpg">
+  <source src="https://cache.ruby-lang.org/pub/media/ruby310_debug_demo.mp4" type="video/mp4">
+</video>
+
+Ruby는 lib/debug.rb를 포함했지만 잘 관리되지 않고 성능과 기능에 여러 이슈가 있었습니다. debug.gem은 lib/debug.rb를 완전히 대체했습니다.
+
+## error_highlight: 백트레이스에서의 세밀한 에러 위치
+
+빌트인된 gem인 error_highlight가 소개되었습니다. 백트레이스에서 세밀한 에러 위치를 포함합니다.
+
+```
+$ ruby test.rb
+test.rb:1:in `<main>': undefined method `time' for 1:Integer (NoMethodError)
+
+1.time {}
+ ^^^^^
+Did you mean?  times
+```
+
+현재는 `NameError`만 지원합니다.
+
+이 gem은 기본적으로 활성화되어있습니다. 명령어에서 `--disable-error_highlight` 옵션을 통해 비활성화할 수 있습니다. [error_highlight 리포지토리](https://github.com/ruby/error_highlight)를 통해 자세한 내용을 참고하세요.
+
+## IRB 자동완성과 문서 표시
+
+IRB는 이제 그냥 코드를 입력하기만 하면 완성 후보 다이얼로그가 나타나는 자동 완성 기능을 포함합니다. 탭과 시프트+탭을 통해 위아래로 이동할 수 있습니다.
+
+만약에 완성 후보를 선택할 때 문서가 설치되어있다면 문서 다이얼로그가 선택한 완성 후보 다이얼로그 옆에 나타나며 해당 내용을 보여줍니다. 알트+d를 눌러서 전체 문서를 읽을 수 있습니다.
+
+<video autoplay="autoplay" controls="controls" muted="muted" width="764" height="510" poster="https://cache.ruby-lang.org/pub/media/ruby310_irb_dialog.png">
+  <source src="https://cache.ruby-lang.org/pub/media/ruby310_irb_dialog.mp4" type="video/mp4">
+</video>
+
+## 다른 주목할만한 새로운 기능들
+
+### 언어
+
+* 해시 리터럴의 값과 키워드 인자를 생략할 수 있습니다. [[Feature #14579]](https://bugs.ruby-lang.org/issues/14579)
+  * `{x:, y:}`는 `{x: x, y: y}`의 편의 문법입니다.
+  * `foo(x:, y:)`는 `foo(x: x, y: y)`의 편의 문법입니다.
+
+* 패턴 매칭에서 핀 연산자 (`^`)에 표현식을 사용할 수 있습니다. [[Feature #17411]](https://bugs.ruby-lang.org/issues/17411)
+
+  ```ruby
+  Prime.each_cons(2).lazy.find_all{_1 in [n, ^(n + 2)]}.take(3).to_a
+  #=> [[3, 5], [5, 7], [11, 13]]
+  ```
+
+* 한 줄 패턴 매칭에서 괄호를 생략할 수 있습니다. [[Feature #16182]](https://bugs.ruby-lang.org/issues/16182)
+
+  ```ruby
+  [0, 1] => _, x
+  {y: 2} => y:
+  x #=> 1
+  y #=> 2
+  ```
+
+### RBS
+
+RBS는 Ruby 프로그램의 구조를 설명하기 위한 언어입니다. 자세한 설명은 [RBS 리포지토리](https://github.com/ruby/rbs)를 참고하세요.
+
+Ruby 3.0.0 이후의 업데이트:
+
+* 제네릭 타입 파라미터를 제한할 수 있습니다. ([PR](https://github.com/ruby/rbs/pull/844))
+* 제네릭 타입 별명을 지원합니다. ([PR](https://github.com/ruby/rbs/pull/823))
+* `rbs collection`이 gem들의 RBS들을 관리하기 위해 소개되었습니다. ([doc](https://github.com/ruby/rbs/blob/master/docs/collection.md))
+* 기본으로 제공되는 표준 라이브러리들의 서명이 추가되어나 갱신되었습니다.
```

Comment:
```suggestion
* 빌트인 및 표준 라이브러리의 시그니처가 추가되거나 갱신되었습니다.
```

Reply:
By which criteria is "시그니처" better than "서명"? Since we use the word "서명" often, e.g., 디지털 서명, I thought that this would be better.
I'm trying to catch up with the translation criteria :D

Reply:
I thought "함수 서명" can be understood as function signing instead of function signature. I googled two words, and "함수 서명" tends to be appeared in machine-translated web pages.

I've just read this once again, and either "... 라이브러리의 함수 서명" or "... 라이브러리의 함수 시그니처" would be fine, but I think "... 라이브러리의 서명" can be understood as certification, signing or something.

Reply:
You're right. I didn't think deeply and thought this signature was some kind of digital signature instead of function signatures. My bad!
I also looked up some [references](https://developer.mozilla.org/ko/docs/Glossary/Signature/Function) that uses the word "함수 시그니처"